app/controllers/state_file/questions/az_retirement_income_subtraction_controller.rb

```diff
@@ -1,6 +1,33 @@
 module StateFile
   module Questions
     class AzRetirementIncomeSubtractionController < RetirementIncomeSubtractionController
+
+      def self.show?(intake)
+        Flipper.enabled?(:show_retirement_ui) && has_eligible_1099rs?(intake)
+      end
+
+      def self.has_eligible_1099rs?(intake)
+        intake.state_file1099_rs.any? do |form1099r|
+          form1099r.taxable_amount&.to_f&.positive?
+        end
+      end
```

Comment:
I did a similar thing in my current PR for ID intakes (in code review) but I think we should move these methods into az/md intakes.

Maybe the `StateFileBaseIntake` should have a method called `eligible_1099rs` which is all 1099rs if we don't need to check specific eligibility, then in the `StateFileAzIntake` (and MD) we define a different `eligible_1099rs` method?

then here we don't have to override the `num_items` or `load_item` methods, rather update the `RetirementIncomeSubtractionController` to utilize `eligible_1099rs` instead of `current_intake.state_file1099_rs`

Reply:
great idea